### PR TITLE
Add pinocchio_python_parser target

### LINKED ISSUE #2474: `pinocchio::python::buildModel` is not build in the main library
`pinocchio::python::buildModel` allow to load a model from a python scripts.

This function should be build in the C++ library and not in the binding. But right now, it's builded only in the bindings and never exposed.

We should add new pinocchio_python_model library to build this code.

### PR DESCRIPTION
**Pull request by jorisv**
Close #2474 

- [x] Create new target
- [x] Move code in core section
- [x] Deprecate bindings header
- [x] Export symbol with dedicated config.hpp
- [x] Use clear CMake variable name

### CODE REVIEW

File: bindings/python/parsers/python/model.cpp [hunk 1]
```diff
@@ -28,11 +26,6 @@ namespace pinocchio
       // Get a dict for the global namespace to exec further python code with
       bp::dict globals = bp::extract<bp::dict>(main_module.attr("__dict__"));
 
-      // We need to link to the pinocchio PyWrap. We delegate the dynamic loading to the python
-      // interpreter.
-      bp::object cpp_module(
-        (bp::handle<>(bp::borrowed(PyImport_AddModule("libpinocchio_pywrap")))));
```
Comment (jorisv) [inline bindings/python/parsers/python/model.cpp:34]: @jcarpent  I removed this part and I don't see the point to use it. Tests are working fine on Linux and OSX (Windows is still running)

File: bindings/python/parsers/python/model.cpp [hunk 2]
```diff
@@ -67,10 +60,6 @@ namespace pinocchio
 
       // close the interpreter
       // cf. https://github.com/numpy/numpy/issues/8097
-#if PY_MAJOR_VERSION < 3
-      Py_Finalize();
-#else
```
Comment (jorisv) [inline bindings/python/parsers/python/model.cpp:72]: @jcarpent I removed Python2 specific code

File: bindings/python/parsers/python/model.cpp [hunk 3]
```diff
@@ -28,11 +26,6 @@ namespace pinocchio
       // Get a dict for the global namespace to exec further python code with
       bp::dict globals = bp::extract<bp::dict>(main_module.attr("__dict__"));
 
-      // We need to link to the pinocchio PyWrap. We delegate the dynamic loading to the python
```
Comment (jorisv) [inline bindings/python/parsers/python/model.cpp:31]: @jcarpent  There is some workaround for BOOST 1.58. Can I remove them ?

File: bindings/python/parsers/python/model.cpp (hunk 2 above)
Comment (jcarpent) [reply bindings/python/parsers/python/model.cpp:72]: Perfect

File: bindings/python/parsers/python/model.cpp (hunk 1 above)
Comment (jcarpent) [reply bindings/python/parsers/python/model.cpp:34]: Fine

File: bindings/python/parsers/python/model.cpp (hunk 3 above)
Comment (jcarpent) [reply bindings/python/parsers/python/model.cpp:31]: Yes, sure.

**Review by jcarpent — APPROVED** [review]

**Review by jcarpent — APPROVED** [review]

**Pull request merged**